### ISSUE TITLE
Restructure repo

### ISSUE THREAD

**Issue by viceice** [zone117x/MimeMapping#96]
We should restructure repo to a more conventional dotnet project

- [ ] add main project to `MimeMapping` folder
- [ ] move projects to `src` and `test` folders ?
- [ ] others ?

**Comment by gintsk** [comment]
Yes, to better align with other .NET projects we can split the root into `src` and `test` with `MimeMapping.sln` in the root folder.

Maybe we should also use the new [artifacts output](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-3/#simplified-output-path) and [global.json](https://learn.microsoft.com/en-us/dotnet/core/tools/global-json)

**Comment by viceice** [comment]
@gintsk do you like to do this?

**Comment by gintsk** [comment]
Yes, I would like to.

**Comment by gintsk** [comment]
Maybe we should rename `Test` project to `MimeMapping.Tests` ?
